assertj-core/src/main/java/org/assertj/core/api/AbstractByteBufferAssert.java

```diff
@@ -0,0 +1,644 @@
+   * assertThat(actual).endsWith(expected);
+   *
+   * // ... but this one fails as the content of "buffer" does not end with the given expected {@code ByteBuffer}.
+   * ByteBuffer expected = ByteBuffer.wrap("test two".getBytes());
+   * assertThat(actual).endsWith(expected);
+   * </code></pre>
+   *
+   * @return {@code this} assertion object.
+   * @throws AssertionError if the actual {@code ByteBuffer} is {@code null}.
+   * @throws AssertionError if the actual {@code ByteBuffer} has not been flipped.
+   * @throws AssertionError if the actual {@code ByteBuffer}s content does not end with the given expected {@code ByeBuffer}.
+   */
+  public SELF endsWith(ByteBuffer expected) {
+    isNotNull();
+    isFlipped();
+
+    byteArrays.assertEndsWith(info, getContent(actual), getContent(expected));
+    return myself;
+  }
+
+  /**
+   * Verifies that the actual {@code ByteBuffer} is marked.
+   *
+   * Example:
+   *
+   * <pre><code class='java'>
+   * ByteBuffer actual = ByteBuffer.wrap("test".getBytes());
+   *
+   * // this assertion succeeds ...
+   * actual.mark();
+   * assertThat(actual).isMarked();
+   *
+   * // ... but this one fails as "buffer" was never marked.
+   * assertThat(actual).isMarked();
+   * </code></pre>
+   *
+   * @return {@code this} assertion object.
+   * @throws AssertionError if the actual {@code ByteBuffer} is {@code null}.
+   * @throws AssertionError if the actual {@code ByteBuffer} is not marked.
+   */
+  public SELF isMarked() {
+    isNotNull();
+    try {
+      actual.duplicate().reset();
+    } catch (InvalidMarkException e) {
+      throwAssertionError(shouldBeMarked(actual));
+    }
+    return myself;
+  }
+
+  /**
+   * Verifies that the actual {@code ByteBuffer} is marked at the given expected position.
+   *
+   * Example:
+   *
+   * <pre><code class='java'>
+   * ByteBuffer actual = ByteBuffer.wrap("test".getBytes());
+   * actual.mark();
+   *
+   * // this assertion succeeds ...
+   * assertThat(actual).isMarkedAt(0);
+   *
+   * // ... but this one fails as "buffer" was marked at position 0.
+   * assertThat(actual).isMarkedAt(1);
+   * </code></pre>
+   *
+   * @param expected integer value representing the expected marked position of the buffer.
+   * @return {@code this} assertion object.
+   * @throws AssertionError if the actual {@code ByteBuffer} is {@code null}.
+   * @throws AssertionError if the actual {@code ByteBuffer} is not marked.
+   * @throws AssertionError if the actual {@code ByteBuffer} is marked at a different position than the given expected position.
+   */
+  public SELF isMarkedAt(int expected) {
+    isNotNull();
+    try {
+      int markedPosition = actual.duplicate().reset().position();
+      if (markedPosition != expected) throwAssertionError(shouldBeMarkedAt(expected, markedPosition, actual));
+    } catch (InvalidMarkException e) {
+      throwAssertionError(shouldBeMarked(actual));
+    }
+    return myself;
+  }
```

Comment:
I took the liberty of creating a `isMarkedAt` assertion as well. I figured that asserting the position of the mark could also be interesting, but let me know if you think that brings no real value.